### ISSUE TITLE
I suggest using tquic as it supports more features.

### ISSUE THREAD

**Issue by kaowul** [M0dEx/quincy#81]
I suggest using tquic (https://github.com/tencent/tquic) as it supports more features.

**Comment by M0dEx** [comment]
I have taken a look at `tquic` and I have a couple of problems with it:
- it lacks any benchmarks comparing it to `quinn` (+ `quinn` is not really a limiting factor in terms of features or performance at the moment)
- it lacks support for QUIC datagrams, which are used extensively by `quincy`
- it is backed by Tencent, which is a corporate entity with, at best, shady practices, based in China (yes, the project itself is open-source, but I'd rather avoid this potential attack vector if at all possible

To sum up, I don't think it is a viable replacement for `quinn`, and I do not trust it due to its backing.

**Comment by kaowul** [comment]
TQUIC is currently the only library that supports Multipath, while QUICHE-CF multipath feature is still under development. Choosing a widely-used library ensures more reliable future updates. It is irrational to have a bias against TQUIC simply because it is developed by a Chinese company. I suggest you review the comparison page here: https://tquic.net/docs/further_readings/comparison/

**Comment by kaowul** [comment]
Support for Multipath is crucial, especially for mobile internet. While there are plenty of VPNs with basic features, support for Multipath is limited. The only other project I found is the outdated https://github.com/kaowul/glorytun. Clearly, a VPN project that supports more advanced features and is better suited for the future network environment will be more popular.

**Comment by kaowul** [comment]
Support for Multipath is crucial, especially for mobile internet. While there are plenty of VPNs with basic features, support for Multipath is limited. The only other project I found is the outdated https://github.com/angt/glorytun  Clearly, a VPN project that supports more advanced features and is better suited for the future network environment will be more popular.

**Comment by M0dEx** [comment]
> TQUIC is currently the only library that supports Multipath, while QUICHE-CF multipath feature is still under development. Choosing a widely-used library ensures more reliable future updates. It is irrational to have a bias against TQUIC simply because it is developed by a Chinese company. I suggest you review the comparison page here: https://tquic.net/docs/further_readings/comparison/

While I understand multipath support would be a nice-to-have, there are other, higher priority tasks at the current moment (route/DNS configuration, additional authentication modules).

As it stands right now, multipath is not yet standardised as a part of the QUIC spec, which is a reason most libraries do not support it. The network path migration currently supported by `quinn` is sufficient to provide a good user experience for the moment.

On the topic of updates, `quinn` is both more widely used and more actively maintained than `tquic`, all while not being dependent on any one maintainer or company to be up-to-date. It also uses a 3rd party audited pure-Rust implementation of the TLS protocol, `rustls`.

To address your last point, I did not reject `tquic` simply due to being backed by a Chinese company. I did it because it is backed by *Tencent*, which is a provably shady conglomerate with ties to the Communist Chinese Party, which is a potential (and very serious) attack vector on the supply chain side of Quincy.
